cookbooks/aws-parallelcluster-install/recipes/c_states.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-install
+# Recipe:: c_states
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return unless node['kernel']['machine'] == 'x86_64'
+
+regen_grub_command = value_for_platform(
+  %w(centos amazon) => { 'default' => '/usr/sbin/grub2-mkconfig -o /boot/grub2/grub.cfg' },
+  'ubuntu' => { 'default' => '/usr/sbin/update-grub' }
+)
+
+# Utility function add an attribute in GRUB_CMDLINE_LINUX_DEFAULT if it is not present
+def append_if_not_present_grub_cmdline(attributes)
+  ruby_block 'Append GRUB_CMDLINE_LINUX_DEFAULT if it do not exist in /etc/default/grub' do
+    block do
+      command_out = shell_out('grep "^GRUB_CMDLINE_LINUX_DEFAULT=" /etc/default/grub').stdout
+
+      if command_out.include? "GRUB_CMDLINE_LINUX_DEFAULT="
+        Chef::Log.debug('Found GRUB_CMDLINE_LINUX_DEFAULT line')
+      else
+        Chef::Log.warn('GRUB_CMDLINE_LINUX_DEFAULT not found - Adding')
+        command = 'echo \'GRUB_CMDLINE_LINUX_DEFAULT=""\' >> /etc/default/grub'
```

Comment:
minor: you can avoid the definition of the command variable if you're using it in a single place.
```ruby
shell_out('echo \'GRUB_CMDLINE_LINUX_DEFAULT=""\' >> /etc/default/grub')
```